src/nvim/eval.c

```diff
@@ -14456,11 +14456,11 @@ static void f_setreg(typval_T *argvars, typval_T *rettv)
   char_u      *strregname;
   char_u      *stropt;
   bool append = false;
-  char_u yank_type;
+  MotionType yank_type;
   long block_len;
 
   block_len = -1;
-  yank_type = MAUTO;
+  yank_type = kUnknownMotion;
```

Comment:
Dunno,  "when motion_type == kLineWise, ..." in a doc/comment (prose) reads naturally as "when the motion type is linewise" and is at the same time clear with how the condition looks in code. I can add  MT if the style mandates it but I don't see the value (what usage below is unclear) and why then are we already adding enums that don't follow it.